### ISSUE TITLE
CommandBarFlyout keyboard navigation broken on RNW Islands app

### ISSUE THREAD

**Issue by rozele** [microsoft/react-native-windows#8306]
## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Run playground-win32
2. Open RNTester
3. Open a context menu on the TextInput for RNTester
4. I cannot navigate through the context menu using the keyboard (via arrow keys or tab).

## Expected Results

I should be able to use the arrow keys or tab to navigate through context menu items.

## Notes
1. This only occurs on CommandBarFlyout (MenuFlyout seems to work fine)
2. This only occurs on XAML Islands in the context of a react-native-windows app (cannot repro on a "standard" XAML islands app)
I considered filing this bug on the microsoft/microsoft-ui-xaml repo, but I am only able to reproduce the issue on react-native-windows.

## Snack, code example, screenshot, or link to a repository:

In the video below, pressing Up/Down arrow keys does not allow navigation through the CommandBarFlyout items (pressing the right arrow on proofing menu flyout only seems to work is when we apply #8249:

https://user-images.githubusercontent.com/1106239/126676232-4dcbb75c-d1a6-4896-916d-38eeed4fc190.mp4

For context, here is the CommandBarFlyout working in a "standard" XAML Islands app:

https://user-images.githubusercontent.com/1106239/126676269-aa866996-a3dd-4583-9855-0901fe4f8e5f.mp4

**Comment by chrisglein** [comment]
Let's reproduce this behavior in an RNW islands app and debug through what's going wrong.

**Comment by rectified95** [comment]
@rozele I'm not seeing the the 'Proofing' command, and the secondary sub-menu: did you customize the textbox in some way?

**Comment by lyahdav** [comment]
> @rozele I'm not seeing the the 'Proofing' command, and the secondary sub-menu: did you customize the textbox in some way?
Did you set the TextInput spellCheck property to true? (from https://github.com/microsoft/react-native-windows/blob/4aa8b3a3e47a5cc0e20a82c9c42ba56487407bfd/vnext/Microsoft.ReactNative/Views/TextInputViewManager.cpp#L640). But I'm pretty sure this issue repros for Eric even if there is no Proofing submenu. Are you able to navigate that context menu with keyboard?

**Comment by rozele** [comment]
@rectified95 - I did not customize the TextBox. Did you force a typo into the TextBox? I still see the Proofing menu on latest main branch in Playground-win32

**Comment by lyahdav** [comment]
I was finally able to repro this consistently, but I don't know what changed that led to the bug starting to appear (I did not upgrade Windows 10 version). But, I was then able to root cause it. The issue seems to happen only to some users (not specific version of Windows 10). And I can fix the issue if I comment out this code in Application.manifest which sets `<dpiAwareness>PerMonitorV2</dpiAwareness>`:
https://github.com/microsoft/react-native-windows/blob/main/packages/playground/windows/playground-win32/Application.manifest#L34-L38

So I think this is actually a XAML bug, not RNW. But I noticed that when I created an Islands app from the VS template it didn't have that `dpiAwareness` in the Application.manifest, so I'm wondering if it makes sense to keep it in the RNW repo given this bug.

**Comment by lyahdav** [comment]
Now I know why the issue started happening to me. I started using a different monitor. Makes sense as this is a per-monitor DPI setting.

**Comment by lyahdav** [comment]
I was curious about other values for `dpiAwareness` and how it affects this bug so I tested them all:
dpiAwareness | Bug present?
------------ | -------------
Not present | No
unaware | No
system | Yes
PerMonitor | Yes
PerMonitorV2 | Yes

**Comment by lyahdav** [comment]
@jonthysell can you try with DPI scaling? And also this bug is only present when there are multiple menu items. I think it's expected behavior that pressing down selects the 2nd item. When the bug happens even that doesn't work.

**Comment by jonthysell** [comment]
@lyahdav  well, it would make sense that pressing down selected the second item, if the first item was actually selected. nothing is visibly selected when using right-click, though the first item **is** when using the context-menu button. so there's still problems even _without_ DPI scaling.

**Comment by lyahdav** [comment]
Using CommandBarFlyout from WinUI 2 fixes this issue but leads to a crash when the flyout is closed. Here's a fork with that change: https://github.com/lyahdav/react-native-windows/tree/win-ui-2-cbf

**Comment by jonthysell** [comment]
This seems to have been fixed by switching to the WinUI CommandBarFlyout and the "new" crash was fixed in 2.7 prerelease.

Do we want to switch to WinUI for the flyout as per this change: https://github.com/lyahdav/react-native-windows/commit/a36d84c6ddc1e26519525022a0389f427bf65263

**Comment by jonthysell** [comment]
Seeing as we just upgraded to WinUI 2.6 for RNW 0.66, which is already in preview and almost ready, I think it's a little late to jump again to 2.7 without more testing. @chrisglein can you move this to 0.67 if you agree?

**Comment by lyahdav** [comment]
@jonthysell I think we should switch to the WinUI 2 CommandBarFlyout for Text and TextInput context menu to fix this bug. I was eventually going to PR this, but it might not be for a while, in case someone else can get to it sooner.

**Comment by jonthysell** [comment]
Do you mean like this? I started this PR here based on what you did: https://github.com/microsoft/react-native-windows/pull/8723

I'm also working on a PR to upgrade us from MUX 2.6 to 2.7, but that might not make RNW 0.66.

**Comment by lyahdav** [comment]
@jonthysell yes, and as you said in that PR, you should wait until the MUX 2.7 upgrade before landing it.

**Comment by chrisglein** [comment]
> Seeing as we just upgraded to WinUI 2.6 for RNW 0.66, which is already in preview and almost ready, I think it's a little late to jump again to 2.7 without more testing. @chrisglein can you move this to 0.67 if you agree?

Agreed. It seems too late to switch from 2.6 to 2.7. Let's target that jump for 0.67 so we can ensure it's stable. In the meantime any individual app can bump to 2.7 on their own.
